### PR TITLE
Fix linter and tests

### PR DESCRIPTION
**Pull request by kkoscielniak**
This PR fixes:
* linter issues, mainly removal of deprecated `componentWillMount` function for `Permissible` component
* test suite, as after removal of the `examples` directory the tests were trying to use a component which was not there already. I've added a missing component to `test/` directory

### CODE REVIEW

File: lib/react-permissible.js [hunk 1]
```diff
@@ -1 +1,6 @@
...
...eof Date)return"date";if(t instanceof RegExp)return"regexp"}return e}function j(t){var e=w(t);switch(e){case"array":case"object":return"an "+e;case"boolean":case"date":case"regexp":return"a "+e;default:return e}}function P(t){return t.constructor&&t.constructor.name?t.constructor.name:k}var E="function"==typeof Symbol&&Symbol.iterator,T="@@iterator",k="<<anonymous>>",S={array:l("array"),bool:l("boolean"),func:l("function"),number:l("number"),object:l("object"),string:l("string"),symbol:l("symbol"),any:function(){return p(r)}(),arrayOf:y,element:function(){function e(e,n,r,o,i){var u=e[n];if(!t(u)){return new s("Invalid "+o+" `"+i+"` of type `"+O(u)+"` supplied to `"+r+"`, expected a single ReactElement.")}return null}return p(e)}(),instanceOf:v,node:function(){function t(t,e,n,r,o){return g(t[e])?null:new s("Invalid "+r+" `"+o+"` supplied to `"+n+"`, expected a ReactNode.")}return p(t)}(),objectOf:h,oneOf:d,oneOfType:b,shape:_,exact:m};return s.prototype=Error.prototype,S.checkPropTypes=u,S.PropTypes=S,S}}).call(e,n(3))},function(t,e,n){"use strict";function r(t){if(null===t||void 0===t)throw new TypeError("Object.assign cannot be called with null or undefined");return Object(t)}/*
```
Comment (piotr-s-brainhub) [inline lib/react-permissible.js]: proposition for further refactor:

this file should be automatically generated before publishing in CircleCI

CC @mdrazyk

File: lib/react-permissible.js [hunk 2]
```diff
@@ -1 +1,6 @@
...
+object-assign
+(c) Sindre Sorhus
```
Comment (piotr-s-brainhub) [inline lib/react-permissible.js]: why Sindre Sorhus?

Comment (kkoscielniak) [reply lib/react-permissible.js]: Not sure to be honest. Probably some external library is "copyrighted" by @sindresorhus 😉

File: lib/react-permissible.js (hunk 1 above)
Comment (kkoscielniak) [reply lib/react-permissible.js]: Agreed.

File: src/components/permissible.js [hunk 3]
```diff
@@ -20,7 +20,9 @@ export function Permissible(
       history: PropTypes.object, // eslint-disable-line react/forbid-prop-types
     };
 
-    componentWillMount() {
+    constructor(props) {
```
Comment (damian-brainhub) [inline src/components/permissible.js]: can you elaborate a little bit why you changed `componentWillMount` to `constructor`? Wouldn't be `componentDidMount` more appropriate? What do you think?

Comment (damian-brainhub) [reply src/components/permissible.js]: oh right, I see what you mean. Okay then

Comment (szymonbrainhub) [reply src/components/permissible.js]: Just to clarify, a component will run method `componentDidMount` even if render returns just `null`. Also fact that `constructor` has been called doesn't mean that it was rendered.

Sandbox:
https://codesandbox.io/s/0oo4583qr0

Comment (kkoscielniak) [reply src/components/permissible.js]: Thanks for the input @szymonbrainhub

**Comment by kkoscielniak** [comment]
There are 4 approvals by Brainhub's Open Source Crew, merging.

**Pull request merged**